Source: revert of upstream commit
Upstream: reported, https://www.spinics.net/lists/linux-btrfs/msg60904.html

From 7c6544d2255922a9a44d778e45875ffbfc60f407 Mon Sep 17 00:00:00 2001
From: Benedikt Morbach <benedikt.morbach@googlemail.com>
Date: Mon, 28 Nov 2016 21:46:17 +0100
Subject: [PATCH] Revert "btrfs-progs: send-stream: check number of read bytes
 from stream"

This reverts commit fefbab75208e570b26b45481db3e5ced6e9e0853.
---
 send-stream.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/send-stream.c b/send-stream.c
index 920c742..328d3a7 100644
--- a/send-stream.c
+++ b/send-stream.c
@@ -67,19 +67,12 @@ static int read_buf(struct btrfs_send_stream *sctx, char *buf, size_t len)
 		}
 		if (rbytes == 0) {
 			ret = 1;
-			goto out_eof;
+			goto out;
 		}
 		pos += rbytes;
 	}
-	ret = 0;
 
-out_eof:
-	if (pos < len) {
-		error("short read from stream: expected %zu read %zu", len, pos);
-		ret = -EIO;
-	} else {
-		sctx->stream_pos += pos;
-	}
+	ret = 0;
 
 out:
 	return ret;
-- 
2.10.2

